### ISSUE TITLE
Improve the Performance of Data Unifying Script

### ISSUE THREAD

**Issue by grayJiaaoLi** [amosproj/amos2024ss08-cloud-native-llm#35]
## User story
1. As a developer
2. I want/need to make the data transformation and unifying script faster.
3. So that we can improve the performance of the whole process. 

## Acceptance criteria
* Reduce the loops in the script
* Reduce the runtime of the unifying script
* Improve the general performance of the Q&A Pairs Transformation and Unification

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project-specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner-approved features
* All tests are passing
* Developers agreed to release
